client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -1670,6 +1670,11 @@ public ReduceFileGroups updateFileGroup(int shuffleId, int partitionId)
       throws CelebornIOException {
     Tuple2<ReduceFileGroups, String> fileGroupTuple =
```

Comment:
```suggestion
    Tuple2<ReduceFileGroups, String> fileGroupTuple =
            reduceFileGroupsMap.compute(shuffleId, (id, existsTuple) -> {
              if (existsTuple == null || existsTuple._1 == null) {
                return loadFileGroupInternal(shuffleId);
              } else {
                return existsTuple;
              }
            });
```

Reply:
I think it would be a little better for the update to be done by the next task. The code would be more concise. But this is acceptable as well.